### PR TITLE
simplify gateway config

### PR DESCRIPTION
**Pull request by akihikokuroda**
### Summary
fix: #326

### Details and comments
There are not a lot of things to simplify the gateway config so I moved most of configs that default values work for into the gateway/values.yaml to make the quantum-serverless/values.yaml and values-ibm.yaml simple.  I also add some comments in gateway/values.yam file.

### CODE REVIEW

**Comment by Tansito** [comment]
is it possible to wait for #928 before close this? I will try to have it for today 🙏

**Comment by Tansito** [comment]
@akihikokuroda #928 probably introduced some problems in the branch, ping me if you need to solve any problem! 🙏

File: charts/quantum-serverless/charts/gateway/templates/deployment.yaml [hunk 1]
```diff
@@ -167,8 +165,8 @@ spec:
             - name: DATABASE_PASSWORD
               valueFrom:
                 secretKeyRef:
-                  name: {{ .Values.secrets.servicePsql.name }}
-                  key: {{ .Values.secrets.servicePsql.key.databasePassword }}
+                  name: postgresql
+                  key: password
```
Comment (Tansito) [inline charts/quantum-serverless/charts/gateway/templates/deployment.yaml:169]: I don't know if this is due to the rebase @akihikokuroda but this change needs to remain as it is.

File: charts/quantum-serverless/charts/gateway/templates/deployment.yaml [hunk 2]
```diff
@@ -280,8 +278,8 @@ spec:
...
```
Comment (Tansito) [inline charts/quantum-serverless/charts/gateway/templates/deployment.yaml:282]: Same as my previous comment

File: charts/quantum-serverless/charts/gateway/values.yaml [hunk 3]
```diff
@@ -3,44 +3,60 @@
...
+  # gateway site url - http://<fullnameOverride>:<service.port>
+  siteHost: "http://gateway:8000"
+  # authentification setting
   auth:
     token: {}
     keycloak: {}
-  superuser:
-    enable: true
+  # django secret key - any secret string
+  django:
```
Comment (Tansito) [inline charts/quantum-serverless/charts/gateway/values.yaml:23]: This key is not needed now, it goes in the secrets part.

File: charts/quantum-serverless/values-ibm.yaml [hunk 4]
```diff
@@ -34,17 +34,7 @@ ingress:
```
Comment (Tansito) [inline charts/quantum-serverless/values-ibm.yaml:34]: Now that we finished with the integration with the infrastructure team I think we could remove this file.

File: charts/quantum-serverless/values.yaml [hunk 5]
```diff
@@ -53,51 +43,6 @@ gateway:
...
-      nodeImage: "icr.io/quantum-public/quantum-serverless-ray-node:0.6.0-py39"
-      opensslImage: registry.access.redhat.com/ubi8/openssl:8.8-7
-      kubectlImage: alpine/k8s:1.28.1@sha256:c37a54b738e4d26f88d4a4d28680bb74dfea1d0e756f039dad900306abc837a8
-    limits:
-      maxJobsPerUser: 2
-      maxComputeResources: 4
-  cos:
-    claimName: gateway-claim
-
-  secrets:
```
Comment (Tansito) [inline charts/quantum-serverless/values.yaml:69]: I think you will need to promote this change to `gateway/values.yaml`.

**Review by Tansito — CHANGES_REQUESTED** [review]

File: charts/quantum-serverless/values.yaml (hunk 5 above)
Comment (akihikokuroda) [reply charts/quantum-serverless/values.yaml:69]: Yes, they are already there except `service`.  The `service` here is in the wrong place.

File: charts/quantum-serverless/charts/gateway/templates/deployment.yaml (hunk 1 above)
Comment (akihikokuroda) [reply charts/quantum-serverless/charts/gateway/templates/deployment.yaml:169]: For the local install, the secret for the password is created by the database install.  So we need a little tweak.  I added another value `databaseSecretName` to separate the name of the secret and the secret name for the password.

File: charts/quantum-serverless/charts/gateway/values.yaml (hunk 3 above)
Comment (akihikokuroda) [reply charts/quantum-serverless/charts/gateway/values.yaml:23]: I see. I take it out.

File: charts/quantum-serverless/values-ibm.yaml (hunk 4 above)
Comment (akihikokuroda) [reply charts/quantum-serverless/values-ibm.yaml:34]: OK.  I remove it.  Thanks!

**Comment by Tansito** [comment]
@akihikokuroda what should we do with this pull request? Should we close it or do you want to solve the conflicts and that I take a look?

**Pull request closed without merging**